controlplane/kubeadm/api/v1alpha4/kubeadmcontrolplanetemplate_webhook.go

```diff
@@ -50,6 +51,15 @@ var _ webhook.Validator = &KubeadmControlPlaneTemplate{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
 func (r *KubeadmControlPlaneTemplate) ValidateCreate() error {
+	// NOTE: KubeadmControlPlaneTemplate is behind ClusterTopology feature gate flag; the web hook
+	// must prevent creating new objects in case the feature flag is disabled.
+	if !feature.Gates.Enabled(feature.ClusterTopology) {
```

Comment:
Probably not worth the effort, if you want you could add a unit test for it. Feel free to just resolve

Reply:
Added 2 simple tests for feature gate enabled and disabled.